cpp/src/integrations/cudf/utils.cpp

```diff
@@ -37,6 +45,43 @@ struct str_cudf_column_scalar_fn {
         RAPIDSMPF_FAIL("not implemented");
     }
 };
+
+struct cudf_column_data_size_fn {
+    template <typename T>
+        requires(cudf::is_fixed_width<T>())
+    size_t operator()(cudf::column_view const& col, rmm::cuda_stream_view) {
+        return static_cast<size_t>(col.size()) * cudf::size_of(col.type())
+               + bitmask_size(col);
+    }
+
+    // string type specialization
+    template <typename T>
+        requires(std::is_same_v<T, cudf::string_view>)
+    size_t operator()(cudf::column_view const& col, rmm::cuda_stream_view stream) {
+        cudf::strings_column_view sv(col);
+        return static_cast<size_t>(sv.chars_size(stream)) + bitmask_size(col);
+    }
+
+    // compound type specialization except string
+    template <typename T>
+        requires(!std::is_same_v<T, cudf::string_view> && cudf::is_compound<T>())
+    size_t operator()(cudf::column_view const& col, rmm::cuda_stream_view) {
+        // compound types (except string) ie. list, dict, structs dont have a
+        // content::data buffer. Data is stored in children columns. So, just return the
+        // bitmask size.
+        return bitmask_size(col);
+    }
+
+    template <typename T>
+    size_t operator()(cudf::column_view const& col, rmm::cuda_stream_view) {
+        RAPIDSMPF_FAIL("not implemented for type: " + cudf::type_to_name(col.type()));
+    }
+
+    static size_t bitmask_size(cudf::column_view const& col) {
+        return col.has_nulls() ? cudf::bitmask_allocation_size_bytes(col.size()) : 0;
```

Comment:
```suggestion
        return col.nullable() ? cudf::bitmask_allocation_size_bytes(col.size()) : 0;
```

Reply:
Ah! TIL. I was looking at [`column::nullable` and it didn't really give me that impression](https://docs.rapids.ai/api/libcudf/stable/classcudf_1_1column#abb4380659f07daa47db1892b7d02a978). 

Anyways, fixed it here #450